### PR TITLE
Add samesite as a session cookie option

### PR DESCRIPTION
**Pull request by tomfotherby**
In PHP 7.3 the [session_set_cookie_params](https://www.php.net/manual/en/function.session-set-cookie-params.php) function had the following change:

> 7.3.0 | An alternative signature supporting an options array has been added. **This signature supports also setting of the `SameSite` cookie attribute**.

This PR brings Yii up to date for PHP 7.3 and implements a workaround for PHP <7.3, ported from Yii2 (ref: https://github.com/yiisoft/yii2/commit/975937e531b488864047f306c047a2240d3bab59#diff-3161fa307a8828ca4cfb175b6bba7838)

This PR is a follow-up to https://github.com/yiisoft/yii/pull/4275 (_add setcookie $options for PHP 7.3 compatibility_)

This PR was first opened as https://github.com/yiisoft/yii/pull/4304 but needed some fixes that required a new PR.

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

**Review by samdark — APPROVED** [review]
Looks OK. Please add/correct CHANGELOG line.

**Comment by tomfotherby** [comment]
Thanks @samdark . I added the CHANGELOG, but now the PR conflicts. This must be a common situation, can you advice what I should do? Do I Rebase on master? I think this would make this PR invalid though, so should I actually create a new PR with the CHANGELOG update done from the beginning?

**Comment by samdark** [comment]
You can merge master branch from this repository into your pull request branch.

**Comment by tomfotherby** [comment]
@samdark - Thanks for the guideance. Done. But now there is a merge commit. Let me know if you need me to rebase for a cleaner history.

**Comment by samdark** [comment]
I'm rebasing at merge so that's not a problem. Thanks!

File: framework/web/CHttpSession.php [hunk 1]
```diff
@@ -250,7 +250,17 @@ public function setCookieParams($value)
 		extract($data);
 		extract($value);
 		$this->freeze();
-		if(isset($httponly))
+		if(isset($httponly) && isset($sameSite))
```
Comment (marcovtwout) [inline framework/web/CHttpSession.php:253]: $sameSite should be lowercase here and below, I'll open a PR to fix it.

Comment (marcovtwout) [reply framework/web/CHttpSession.php:253]: https://github.com/yiisoft/yii/pull/4313

Comment (tomfotherby) [reply framework/web/CHttpSession.php:253]: Thank you for catching this. Sorry!

**Pull request merged**